dagshub/mlflow/patch.py

```diff
@@ -0,0 +1,328 @@
+            for index, attr_name in enumerate(attrs):
+                parent_attr = attr
+                attr = getattr(parent_attr, attr_name)
+                if index == len(attrs) - 1:
+                    setattr(parent_attr, attr_name, unpatched_func)
+
+        self.patched_modules.clear()
+        self.original_func_lookup.clear()
+
+
+_global_mlflow_patch: Optional[MlflowMonkeyPatch] = None
+_patch_mutex = threading.Lock()
+
+
+def _patch_mlflow(funcs_to_patch: List[str], classes_to_patch: List[str], guaranteed_raises: List[str]):
+    import mlflow
+
+    global _global_mlflow_patch
+
+    with _patch_mutex:
+        if _global_mlflow_patch is not None:
+            logger.warning("MLflow is already patched")
+            return
+
+        logger.warning(
+            "Patching MLflow to prevent any MLflow exceptions from being raised. "
+            "Call dagshub.mlflow.unpatch_mlflow() to undo"
+        )
+
+        patch = MlflowMonkeyPatch(funcs_to_patch, classes_to_patch, guaranteed_raises)
+        patch.monkey_patch_module(mlflow)
+        _global_mlflow_patch = patch
+
+
+def _resolve_patches(
+    include: Optional[List[str]],
+    exclude: Optional[List[str]],
+    include_classes: Optional[List[str]],
+    patch_top_level: bool,
+    patch_mlflow_client: bool,
+    raise_on_log_model: bool,
+) -> Tuple[List[str], List[str], List[str]]:
+    """
+    Resolve the arguments for patch_mlflow
+
+    Returns:
+        Tuple of:
+        - List of functions that need to be patched
+        - List of classes that need to be patched
+        - List of functions that should raise exceptions if they are in the stack
+    """
+    funcs_to_patch = _default_patch_functions.copy()
+    classes_to_patch = _default_patch_classes.copy()
+    guaranteed_raises = _default_guaranteed_raises.copy()
+
+    if not patch_top_level:
+        for f in _top_level_functions:
+            if f in funcs_to_patch:
+                funcs_to_patch.remove(f)
+
+    if not patch_mlflow_client:
+        for f in _mlflow_client_functions:
+            if f in funcs_to_patch:
+                funcs_to_patch.remove(f)
+        classes_to_patch.remove("mlflow.tracking.client.MlflowClient")
+
+    if include is not None:
+        funcs_to_patch = funcs_to_patch + include
+
+    if include_classes is not None:
+        classes_to_patch = classes_to_patch + include_classes
+
+    if exclude is not None:
+        for f in exclude:
+            if f in funcs_to_patch:
+                funcs_to_patch.remove(f)
+
+    if not raise_on_log_model:
+        if "log_model" in guaranteed_raises:
+            guaranteed_raises.remove("log_model")
+
+    return funcs_to_patch, classes_to_patch, guaranteed_raises
+
+
+def patch_mlflow(
+    include: Optional[List[str]] = None,
+    exclude: Optional[List[str]] = None,
+    include_classes: Optional[List[str]] = None,
+    raise_on_log_model=True,
+    patch_top_level=True,
+    patch_mlflow_client=False,
```

Comment:
Why is the default here false?

Reply:
Good catch, I set it to False before I did the log_model failure thing and forgot to return it back.
Thank you